### PR TITLE
Refs #22654 -- Add additional tests and remove release notes

### PR DESCRIPTION
**Pull request by raphaelm**
See @timgraham's comments on #8312.

### CODE REVIEW

File: tests/i18n/tests.py [hunk 1]
```diff
@@ -874,6 +874,17 @@ def test_sanitize_separators(self):
                 # Suspicion that user entered dot as decimal separator (#22171)
                 self.assertEqual(sanitize_separators('10.10'), '10.10')
 
+        with self.settings(USE_L10N=False, DECIMAL_SEPARATOR=','):
+            self.assertEqual(sanitize_separators('1001,10'), '1001.10')
+            self.assertEqual(sanitize_separators('1001.10'), '1001.10')
+
+        with self.settings(USE_L10N=False, DECIMAL_SEPARATOR=',', USE_THOUSAND_SEPARATOR=True,
```
Comment (timgraham) [inline tests/i18n/tests.py]: We usually use hanging indent:
```
with self.settings(
    USE_L10N=False, DECIMAL_SEPARATOR=',', USE_THOUSAND_SEPARATOR=True,
    THOUSAND_SEPARATOR='.',
):
```

File: tests/i18n/tests.py [hunk 2]
```diff
@@ -874,6 +874,17 @@ def test_sanitize_separators(self):
...
+                           THOUSAND_SEPARATOR='.'):
+            self.assertEqual(sanitize_separators('1.001,10'), '1001.10')
+            self.assertEqual(sanitize_separators('1001,10'), '1001.10')
+            self.assertEqual(sanitize_separators('1001.10'), '1001.10')
+            self.assertEqual(sanitize_separators('1,001.10'), '1.001.10')  # Invalid output, as expected
```
Comment (timgraham) [inline tests/i18n/tests.py]: I think "as expected" can be chopped.

**Comment by raphaelm** [comment]
@timgraham I pushed a new version with your requested changes

File: docs/releases/2.0.txt [hunk 3]
```diff
@@ -305,6 +301,11 @@ Miscellaneous
 * :class:`~django.views.generic.base.RedirectView` no longer silences
   ``NoReverseMatch`` if the ``pattern_name`` doesn't exist.
 
+* :class:`~django.forms.FloatField` and :class:`~django.forms.DecimalField`
+  now respect :setting:`DECIMAL_SEPARATOR` and :setting:`THOUSAND_SEPARATOR`
+  during validation correctly when :setting:`USE_L10N` is off.
```
Comment (timgraham) [inline docs/releases/2.0.txt]: I'm not sure if the note describes the issue in enough detail. (Why is it considered backwards incompatible?) Perhaps giving an example would help? Also "correctly" seems out of place in the sentence. \\cc @claudep

Comment (raphaelm) [inline docs/releases/2.0.txt]: I'm happy for suggestions here @claudep. I don't really see the backwards incompatiblity, the only thing I see is that certain inputs which have been wrongfully rejected as invalid are now accepted.

Comment (claudep) [inline docs/releases/2.0.txt]: Isn't the example I gave in https://code.djangoproject.com/ticket/22654#comment:19 clear enough? Or was I wrong?

Comment (raphaelm) [inline docs/releases/2.0.txt]: Sorry @claudep I got somehow confused and forgot there was an example in your comment. I think the example needs to be a bit different (``LANGUAGE_CODE`` is ignored if ``USE_I10N`` is off), but I proposed one now in the patch.

**Comment by claudep** [comment]
Great, thanks!

**Pull request merged**